goldenmaster/Plugins/ApiGear/Source/ApiGearOLink/Private/OLinkHostConnection.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+THIRD_PARTY_INCLUDES_START
+#include "olink/remoteregistry.h"
+#include "olink/remotenode.h"
+THIRD_PARTY_INCLUDES_END
+#include <UObject/Object.h>
+#include "ApiGearLogCategories.h"
+#include "INetworkingWebSocket.h"
+
+/// @brief an instance of this class is held by the OLink server worker for each existing connection
+class APIGEAROLINK_API FOLinkHostConnection
+{
+
+public:
+	FOLinkHostConnection(INetworkingWebSocket* InSocket, ApiGear::ObjectLink::RemoteRegistry& InRegistry, ApiGear::ObjectLink::WriteLogFunc logFunc)
+		: Socket(InSocket)
+		, Registry(InRegistry)
+		, Node(ApiGear::ObjectLink::RemoteNode::createRemoteNode(InRegistry))
+	{
+		Node->onLog(logFunc);
+		Node->onWrite([InSocket](std::string msg)
```

Comment:
Node->onWrite([InSocket](std::string msg) ->
Node->onWrite([**Socket**]( **const std::string&** msg)

Reply:
done. sticking with insocket, otherwise we have to capture this which I don't want to do here